### ISSUE TITLE
[Craft 5] Blitz Hints not picking up an element queries that are not eager-loaded

### ISSUE THREAD

**Issue by svondervoort** [putyourlightson/craft-blitz#702]
### Bug Report

For example we have an Entries field where you can select multiple entries.
In the template we have logic to get those entries and loop through them like this:
``` twig
{% set blockProducts = entry.blockProducts.all() %}

{% for productEntry in blockProducts %}
    {{ productEntry.title }}
{% endfor %}
</section>
```

In the Craft 4 version Blitz Hints would tell us the eager-load these items using:
``` twig
{% do craft.app.elements.eagerLoadElements(className(entry), [entry], ['blockProducts']) %}
```

But in Craft 5 no hint is triggered. Not sure if this is a change in Craft 5 or Blitz Hints not picking it up?

I would think we would have to eager-load using:
``` twig
{% set blockProducts = entry.blockProducts.eagerly.all() %}
```

Maybe I'm missing some crucial detail or looking into it too much.

#### Application Info

- PHP version: 8.3.8
- Craft edition & version: Pro 5.3.5
- Database driver & version: MariaDB 10.10.7

#### Installed Plugins

- Blitz: 5.6.4
- CKEditor: 4.2.0
- Embedded Assets: 5.1.2
- Enupal Translate: 4.0.2
- Environment Label: 5.0.0
- Formie: 3.0.3
- Google Maps: 5.0.2
- Hyper: 2.0.4
- Image Hotspots: 3.0.1
- Laméco Blitz: 5.0.0
- Multi Translator: 2.7.0
- Navigation: 3.0.3
- Retour: 5.0.3
- SEOmatic: 5.1.2
- Smith: 3.1.0
- Sprig: 3.4.0
- Vite: 5.0.1

#### Loaded Modules

- lameco: modules\lameco\Module
- codeeditor: nystudio107\codeeditor\CodeEditor
- sprig-core: putyourlightson\sprig\Sprig
- verbb-base: verbb\base\Base

#### Blitz Plugin Settings

```json
{
    "debug": false,
    "hintsEnabled": true,
    "cachingEnabled": false,
    "refreshCacheEnabled": true,
    "refreshMode": 0,
    "includedUriPatterns": [
        {
            "siteId": "",
            "uriPattern": ".*"
        }
    ],
    "excludedUriPatterns": [],
    "cacheStorageType": "putyourlightson\\blitz\\drivers\\storage\\FileStorage",
    "cacheStorageSettings": {
        "folderPath": "@webroot\/cache\/blitz",
        "compressCachedValues": true
    },
    "cacheStorageTypes": [],
    "cacheGeneratorType": "putyourlightson\\blitz\\drivers\\generators\\HttpGenerator",
    "cacheGeneratorSettings": {
        "concurrency": 1
    },
    "cacheGeneratorTypes": [],
    "customSiteUris": [],
    "cachePurgerType": "putyourlightson\\blitz\\drivers\\purgers\\DummyPurger",
    "cachePurgerSettings": [],
    "cachePurgerTypes": [],
    "deployerType": "putyourlightson\\blitz\\drivers\\deployers\\DummyDeployer",
    "deployerSettings": [],
    "deployerTypes": [],
    "ssiEnabled": false,
    "ssiTagFormat": "<!--#include virtual=\"{uri}\" -->",
    "detectSsiEnabled": true,
    "esiEnabled": false,
    "queryStringCaching": 1,
    "includedQueryStringParams": [],
    "excludedQueryStringParams": [],
    "apiKey": "",
    "generatePagesWithQueryStringParams": true,
    "purgeAssetImagesWhenChanged": true,
    "refreshCacheAutomaticallyForGlobals": true,
    "refreshCacheWhenElementMovedInStructure": true,
    "refreshCacheWhenElementSavedUnchanged": false,
    "refreshCacheWhenElementSavedNotLive": false,
    "cacheNonHtmlResponses": false,
    "trackElements": true,
    "trackElementQueries": true,
    "excludedTrackedElementQueryParams": [],
    "cacheDuration": null,
    "nonCacheableElementTypes": [],
    "sourceIdAttributes": [],
    "liveStatuses": [],
    "integrations": [
        "putyourlightson\\blitz\\drivers\\integrations\\CommerceIntegration",
        "putyourlightson\\blitz\\drivers\\integrations\\FeedMeIntegration",
        "putyourlightson\\blitz\\drivers\\integrations\\SeomaticIntegration"
    ],
    "defaultCacheControlHeader": "no-store",
    "cacheControlHeader": "public, max-age=31536000",
    "cacheControlHeaderExpired": "public, max-age=5",
    "sendPoweredByHeader": true,
    "outputComments": true,
    "refreshCacheJobPriority": 10,
    "driverJobBatchSize": 100,
    "driverJobPriority": 100,
    "queueJobTtr": 300,
    "maxRetryAttempts": 10,
    "maxUriLength": 255,
    "mutexTimeout": 1,
    "commands": [],
    "injectScriptEvent": "DOMContentLoaded",
    "injectScriptPosition": 3
}
```

#### Recommendations

- ❌ Queue jobs are configured to run automatically via web requests.
- ❌ The blitz/cache/refresh-expired console command has not been executed within the past 24 hours.
- ✅ Blitz is configured not to refresh cached pages when an element is saved but unchanged.
- ✅ Blitz is configured not to refresh cached pages when an element is saved but not live.
- ✅ Image transforms are configured to be generated before page load.
- ✅ No globals exist.
- ✅ The @web alias is not used in the base URL of any sites.
- ✅ The Blitz Hints utility is not reporting any eager-loading opportunities.
- ✅ The Async Queue plugin is not installed or enabled.

#### Site Tracking [1]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [2]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [3]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [4]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [5]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [6]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [7]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [8]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [9]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [10]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [11]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [12]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [13]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [14]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [15]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [16]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [17]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [18]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

#### Site Tracking [19]

- Tracked Pages: 0
- Tracked Includes: 0
- Tracked Query String Params: 0
- Tracked Elements: 0
- Tracked Element Queries: 0
- Tracked Tags: 0

**Comment by bencroker** [comment]
Thanks for reporting this. Compatibility with Craft 5.3 has already been addressed for the upcoming 5.7.0 release of Blitz (likely next week). 
https://github.com/putyourlightson/craft-blitz/blob/develop/CHANGELOG.md

**Comment by svondervoort** [comment]
Great, looking forward to checking off all missed hints :)

**Comment by bencroker** [comment]
Released in [5.7.0](https://github.com/putyourlightson/craft-blitz/releases/tag/5.7.0), so you should start seeing those hints after updating, refreshing the cache and revisiting those pages.

**Comment by svondervoort** [comment]
Yes, just tested and its working again. Thanks!
